addons/hr_holidays/models/hr_leave.py

```diff
@@ -731,48 +733,69 @@ def _check_approval_update(self, state):
         """ Check if target state is achievable. """
         current_employee = self.env['hr.employee'].search([('user_id', '=', self.env.uid)], limit=1)
         is_team_leader = self.env.user.has_group('hr_holidays.group_hr_holidays_team_leader')
+        is_officer = self.env.user.has_group('hr_holidays.group_hr_holidays_user')
         is_manager = self.env.user.has_group('hr_holidays.group_hr_holidays_manager')
+
+        # FIXME This is probably handled by hr_leave_allocation_rule_employee_update ir.rule
+        if state in ['refuse', 'validate1', 'validate'] and not is_team_leader:
+            raise UserError(_('Only a Team Leader, Leave Officer or Manager can approve or refuse leave requests.'))
+
         for holiday in self:
             val_type = holiday.holiday_status_id.validation_type
-            if state == 'confirm':
-                continue
 
-            if state == 'draft':
-                if holiday.employee_id != current_employee and not is_manager:
-                    raise UserError(_('Only a Leave Manager can reset other people leaves.'))
-                continue
+            if not is_manager:
+                if state == 'confirm':
+                    continue
+                elif state == 'draft':
+                    if holiday.employee_id != current_employee:
```

Comment:
Officers should be able too

Reply:
According to the docstring he shouldn't
And that's what was previously implemented

``` python
if state == 'draft':
    if holiday.employee_id != current_employee and not is_manager:
        raise UserError(_('Only a Leave Manager can reset other people leaves.'))
    continue
```